### PR TITLE
Copy attributes from a nonlist to a list and its elements

### PR DESCRIPTION
**Pull request by JustinCCYap**
Required by branch DS-2702 of flipTransformations

### CODE REVIEW

**Comment by mwmclean** [comment]
This looks like a pretty strange and significant change to me. The function was intended to copy between lists of the same length and nested structure. You seem to have broken unit tests in multiple reverse dependencies. You have not updated the documentation for `CopyAttributes`. You should update the details section and the examples section with a useful example. What is the use case for this in Shapley regression?

**Comment by JustinCCYap** [comment]
I have updated the documentation including adding examples. This change was necessary because I was calling CopyAttributes in flipTransformations to copy attributes from a vector to a data frame, which caused a crash in CopyAttributes (https://github.com/Displayr/flipTransformations/blob/a726baf0eb31f83cbb3c1deb23d2e77500a96332/R/variable.R#L167). I was changing flipTransformations in order to address point 1 of your comment in https://numbers.atlassian.net/browse/DS-2702, which was due to the label attribute not  being lost when calling AsNumeric.

**Comment by JustinCCYap** [comment]
Unit tests were failing in some packages that used flipTransformations::AsNumeric after I made changes to flipTransformations in branch DS-2702. This is because they required the DS-2702 branch of flipU to run correctly.

File: R/properties.R [hunk 1]
```diff
@@ -138,22 +138,47 @@ addQuotesOrComma <- function(v, term)
 }
 #' Copy attributes from one object to another
 #'
-#' Copies the "label", "name", "question" and "questiontype" attributes
-#' for each for variable in a \code{\link{data.frame}}.
+#' Copies attributes such as "label", "name", "question" and "questiontype"
+#' from one object to another. If both objects are data frames,
```
Comment (mwmclean) [inline R/properties.R]: How about "If both objects are lists [data frames], elements [columns] in the recipient list [data frame] will also receive attributes from elements [columns] with the same name in the donor list [data frame]."

File: R/properties.R [hunk 2]
```diff
@@ -138,22 +138,47 @@ addQuotesOrComma <- function(v, term)
...
+#' variables with the same name in the donor data frame.
 #' @param data.without.attributes an object to receive attributes from, such as
 #' a data.frame, list, or matrix
 #' @param data.with.attributes an object to copy attributes from
 #' @param attr.to.not.copy character vector of attribute names appearing in
 #' \code{data.with.attributes} that should not be copied
 #' @return A copy of \code{data.without.attributes} with all the attributes
 #' of \code{data.with.attributes}.
-#' @details In the case when both arguments are \code{data.frame}s, any attributes
+#' @details In the case when both arguments are data frames, any attributes
```
Comment (mwmclean) [inline R/properties.R:154]: Can you please also add a note that the function is recursive and will copy attributes when the inputs are nested lists?

Comment (JustinCCYap) [reply R/properties.R:154]: OK done to both your comments

**Review by jrwishart — COMMENTED** [review]
Approved

**Pull request merged**